### DISCUSSION TITLE
Is there any way to optimize PC1?

### DISCUSSION THREAD

**Discussion by nujz** [filecoin-project/rust-fil-proofs#1626] [Q&A]
Is there any way to optimize PC1?

**Comment by hail100** [comment]
https://github.com/storswiftlabs/rust-fil-proofs-mpool/blob/master/storage-proofs-porep/src/stacked/vanilla/create_label/mem_pool_arch.md

**Comment by hail100** [comment]
hi @vmx, I  found PC1 `create_base_merkle_tree`  hash() use of cpu  will affect SDR bind cores, so 32G PC1 tasks will run for one more hour.

**Comment by vmx** [comment]
@hail100: yes only the layer generation is bound to cores. The tree building of `create_base_merkle_tree` isn't bound to any specific cores.

**Comment by hail100** [comment]
Maybe need other project support. https://github.com/filecoin-project/merkletree/blob/master/src/hash.rs

**Reply by hail100** [reply]
https://github.com/supranational/filecoin_pc1
